hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -1319,8 +1318,22 @@ protected boolean areSomeRegionReplicasColocated(Cluster c) {
     return assignments;
   }
 
-  protected Cluster createCluster(List<ServerName> servers, Collection<RegionInfo> regions,
-      boolean hasRegionReplica) {
+  protected Cluster createCluster(List<ServerName> servers, Collection<RegionInfo> regions)
+      throws HBaseIOException {
+    boolean hasRegionReplica = false;
+    try {
+      for (RegionInfo regionInfo : regions) {
+        TableName tableName = regionInfo.getTable();
+        if (services != null &&
+            services.getTableDescriptors().get(tableName).getRegionReplication() > 1) {
```

Comment:
I think the descriptors are anyway cached right? If so good.

Reply:
Updated in new patch. Will only getTableDescriptors once now.